### ISSUE TITLE
Bug: [consistent-type-exports] fixMixedExportsWithInlineTypeSpecifier not always enforced

### ISSUE THREAD

**Issue by romanstetsyk** [typescript-eslint/typescript-eslint#7524]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.2.1-rc&fileType=.ts&code=C4TwDgpgBAKlC8UB2BXAtgIwgJwNwCgBjAeyQGdgoAPALlgSgBYAmA-CKsY7Sgb2oA0UUJHoBfAkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipegA9iHaPlQYA2uGw5E0aJMgAaBYuBQAZpVEBZA4gAmAUXGTpAdUr4AFgEkmjJogAqJRAGVSZSn1ldDBdAEN4ZEQwAF8AXQUYkCSgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```typescript
type T = number;
const x: T = 42;

export { x, type T };
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/consistent-type-exports": [
      "error",
      { "fixMixedExportsWithInlineTypeSpecifier": false }]
  }
}
```

### Expected Result

I expected that there would be an error and the auto-fix would correct the last line of code to 
```
export type { T };
export { x };
``` 

In general, I expect that `{ "fixMixedExportsWithInlineTypeSpecifier": false }` doesn't allow mixing exports with inline type specifiers, as the option name implies.

### Actual Result

No error

**Comment by bradzacher** [comment]
It probably could be built into the [`import/consistent-type-specifier-style`](https://github.com/import-js/eslint-plugin-import/blob/HEAD/docs/rules/consistent-type-specifier-style.md) rule?
